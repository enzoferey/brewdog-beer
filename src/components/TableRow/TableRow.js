import React from "react";
import PropTypes from "prop-types";
import style from "./table-row.scss";

import Button from "components/Button";

const TableRow = ({
  item,
  done,
  duration,
  waiting,
  callback,
  gradientButton,
}) => (
  <div className={style.main}>
    {item}
    <Button
      done={done}
      delay={duration}
      waiting={waiting}
      callback={callback}
      background={gradientButton}
    />
  </div>
);

TableRow.propTypes = {
  item: PropTypes.element,
  ...Button.propTypes,
};

export default TableRow;
