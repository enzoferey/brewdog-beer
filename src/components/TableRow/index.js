export { IconLabel, default } from "./TableRow";
