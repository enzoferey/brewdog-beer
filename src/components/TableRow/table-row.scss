.main {
  border-bottom: 1px solid #707070;
}
