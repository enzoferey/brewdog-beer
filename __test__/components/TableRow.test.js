import React from "react";
import TableRow from "components/TableRow";

const MyText = () => <p>hi</p>;

const props = {
  item: <MyText />,
  done: false,
  duration: 10,
  waiting: false,
  callback: jest.fn(),
  gradientButton: "white",
};

test("renders correctly", () => {
  testSnapshot(<TableRow {...props} />);
});
