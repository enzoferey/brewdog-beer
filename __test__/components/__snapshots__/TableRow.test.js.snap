// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <p>
    hi
  </p>
  <div
    className={undefined}
  >
    <div
      className="undefined "
      onClick={[Function]}
      style={
        Object {
          "background": "white",
        }
      }
    >
      <p
        className={undefined}
      >
        Add
      </p>
      <div
        className="undefined "
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
